tony-history-server/app/controllers/JobEventPageController.java

```diff
@@ -0,0 +1,60 @@
+package controllers;
+
+import cache.CacheWrapper;
+import com.google.common.base.Preconditions;
+import com.google.common.cache.Cache;
+import com.linkedin.tony.TonyConfigurationKeys;
+import com.typesafe.config.Config;
+import hadoop.Configuration;
+import java.util.List;
+import javax.inject.Inject;
+import models.JobEvent;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hdfs.HdfsConfiguration;
+import play.Logger;
+import play.Logger.ALogger;
+import play.mvc.Controller;
+import play.mvc.Result;
+import utils.HdfsUtils;
+
+import static utils.HdfsUtils.*;
+import static utils.ParserUtils.*;
+
+
+public class JobEventPageController extends Controller {
+  private static final ALogger LOG = Logger.of(JobEventPageController.class);
+  private final Config config;
+
+  @Inject
+  public JobEventPageController(Config config) {
+    this.config = config;
+  }
+
+  public Result index(String jobId) {
+    HdfsConfiguration conf = Configuration.getHdfsConf();
+    FileSystem myFs = HdfsUtils.getFileSystem(conf);
+    Cache<String, List<JobEvent>> cache = CacheWrapper.getEventCache();
+
+    if (myFs == null) {
+      return internalServerError("Failed to initialize file system");
+    }
+
+    List<JobEvent> listOfEvents;
+    Path tonyHistoryFolder = new Path(config.getString(TonyConfigurationKeys.TONY_HISTORY_LOCATION));
+    listOfEvents = cache.getIfPresent(jobId);
+    if (listOfEvents != null) {
+      return ok(views.html.event.render(listOfEvents));
+    }
+    List<Path> jobFolder = getJobFolders(myFs, tonyHistoryFolder, jobId);
```

Comment:
You could reduce the number of `return` statements by restructuring as follows:

```
listOfEvents = cache.getIfPresent(jobId);
if (listOfEvents == null) {
  // read from file

  if (listOfEvents.size() == 0) {
    ...
  }
  cache.put(...);
}

return ok(...);
```

Reply:
I guess this is a style thing. I prefer having more return statements to make the code flatter rather than nested code to reduce # of returns.

Reply:
True, this is mainly stylistic. One argument in favor of my approach is less code duplication. In your approach, you duplicate the line `return ok(views.html.event.render(listOfEvents));`. I try to follow the DRY (Don't Repeat Yourself) principle.